front/app/components/EventAttendanceButton/index.tsx

```diff
@@ -140,87 +134,65 @@ const EventAttendanceButton = ({ event }: EventAttendanceButtonProps) => {
   };
 
+  // Button disabled logic
   const maxAttendeesReached =
     event.attributes.maximum_attendees !== null &&
     event.attributes.attendees_count >= event.attributes.maximum_attendees;
-
-  const getButtonText = () => {
-    // First check if maximum attendees is reached and user is not attending
-    if (maxAttendeesReached && !userIsAttending) {
-      return formatMessage(messages.noSpotsLeft);
-    }
-    // TODO: Fix this the next time the file is edited.
-    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    else if (customButtonText && event?.attributes.using_url) {
-      return customButtonText;
-    } else if (userIsAttending) {
-      return formatMessage(messages.attending);
-    }
-    return formatMessage(messages.attend);
-  };
-
-  const getButtonIcon = () => {
-    // Don't show any icon when no spots left
-    if (maxAttendeesReached && !userIsAttending) {
-      return undefined;
-    }
-    // TODO: Fix this the next time the file is edited.
-    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    else if (event?.attributes.using_url) {
-      return undefined;
-    } else if (userIsAttending) {
-      return 'check';
-    }
-    return 'plus-circle';
-  };
-
-  const isLoading = isAddingAttendance || isRemovingAttendance;
-
-  // Permissions disabled reasons
-  const buttonDisabled =
-    (!!disabled_reason && !isFixableByAuthentication(disabled_reason)) ||
-    (maxAttendeesReached && !userIsAttending);
+  const hasProjectDisabledReason =
+    !!disabled_reason && !isFixableByAuthentication(disabled_reason);
+  const buttonDisabled = maxAttendeesReached || hasProjectDisabledReason;
 
   const permissionDisabledMessageDescriptor = getPermissionsDisabledMessage(
     'attending_event',
     disabled_reason
   );
-  const disabledMessage =
-    permissionDisabledMessageDescriptor &&
-    formatMessage(permissionDisabledMessageDescriptor);
+  const disabledMessage = maxAttendeesReached
+    ? formatMessage(messages.maxAttendeesReached)
+    : hasProjectDisabledReason && permissionDisabledMessageDescriptor
+    ? formatMessage(permissionDisabledMessageDescriptor)
+    : null;
+
+  const buttonProps: ButtonProps = {
+    ml: 'auto',
+    width: '100%',
+    iconSize: '20px',
+    onClick: (event) => {
+      event.preventDefault();
+      handleClick();
+    },
+    processing: isAddingAttendance || isRemovingAttendance,
+    className: 'e2e-event-attendance-button',
+  };
 
   return (
     <>
-      <Tooltip
-        disabled={!disabled_reason}
-        placement="bottom"
-        content={disabledMessage}
-        useContentWrapper={false}
-      >
+      {userIsAttending ? (
         <Button
-          ml="auto"
-          width={'100%'}
-          iconPos={userIsAttending ? 'left' : 'right'}
-          icon={getButtonIcon()}
-          iconSize="20px"
-          disabled={buttonDisabled || (maxAttendeesReached && !userIsAttending)}
```

Comment:
Would keep all conditions in one variable, vs. having one condition in the props, like you did here.